File: app/src/main/java/com/elitec/alejotaller/feature/agent/domain/caseUse/SendAgentMessageCaseUse.kt
package com.elitec.alejotaller.feature.agent.domain.caseUse

import com.elitec.alejotaller.feature.agent.domain.entity.AgentMessage
import com.elitec.alejotaller.feature.agent.domain.entity.AgentReply
import com.elitec.alejotaller.feature.agent.domain.repository.AgentRepository

class SendAgentMessageCaseUse(
    private val repository: AgentRepository,
) {
    suspend operator fun invoke(
        text: String,
        history: List<AgentMessage> = emptyList(),
    ): AgentReply {
        val trimmed = text.trim()
        require(trimmed.isNotEmpty()) { "El mensaje no puede estar vacío" }
        return repository.sendMessage(trimmed, history)
    }
}

File: web/src/core/feature/agent/domain/caseuse/SendAgentMessageCaseUse.ts
import type { AgentMessage, AgentReply } from "../entity/AgentMessage";
import type { AgentRepository } from "../repository/agent.repository";

export class SendAgentMessageCaseUse {
  constructor(private readonly repo: AgentRepository) {}

  async execute(input: {
    text: string;
    history?: ReadonlyArray<Pick<AgentMessage, "role" | "content">>;
  }): Promise<AgentReply> {
    const text = input.text?.trim() ?? "";
    if (!text) {
      throw new Error("El mensaje no puede estar vacío");
    }
    return this.repo.sendMessage(text, input.history);
  }
}
